batchflow/models/tf/efficientnet.py

```diff
@@ -0,0 +1,318 @@
+        if kwargs.get('layout'):
+            return cls.block(inputs, name=name, **kwargs)
+        return inputs
+
+    @classmethod
+    def body(cls, inputs, name='body', **kwargs):
+        kwargs = cls.fill_params('body', **kwargs)
+
+        with tf.variable_scope(name):
+            blocks = kwargs.pop('blocks')
+            x = inputs
+            for i, block_args in enumerate(blocks):
+                x = cls.block(x, name='block-%d' % i, **{**kwargs, **block_args})
+
+            return x
+
+    @classmethod
+    def head(cls, inputs, name='head', **kwargs):
+        kwargs = cls.fill_params(name, **kwargs)
+        if kwargs.get('layout'):
+            return cls.block(inputs, name=name, **kwargs)
+        return inputs
+
+    @classmethod
+    def block(cls, inputs, name, **block_args):
+        """ block of repeated layers of same structure"""
+        base_block = block_args.pop('base', None)
+
+        new_layer_args = cls.update_layer_params(block_args)
+        if base_block is None:
+            with tf.variable_scope(name):
+                repeats = new_layer_args.get('repeats', 1)
+                x = inputs
+                for i in range(repeats):
+                    x = conv_block(x, name='layer-%d' % i, **new_layer_args)
+
+                return x
+        else:
+            return base_block(inputs, name=name, **new_layer_args)
+
+    @classmethod
+    def update_layer_params(cls, kwargs):
+        """ calculate filters and number of repeats depending on scaling factors """
+        new_kwargs = dict(**kwargs)
+        scalable = new_kwargs.get('scalable', False)
+
+        if not scalable:
+            return new_kwargs
+
+        w_factor = new_kwargs.get('width_factor')
+        filters = new_kwargs.get('filters')
+        if filters and w_factor > 1:
+            if isinstance(filters, int):
+                new_kwargs['filters'] = cls.round_filters(filters, w_factor)
+            elif isinstance(filters, list):
+                new_kwargs['filters'] = [cls.round_filters(f, w_factor) for f in filters]
+            else:
+                raise ValueError("filters should be int or list, {} given".format(type(filters)))
+
+        d_factor = kwargs.get('depth_factor')
+        repeats = kwargs.get('repeats')
+        if repeats and d_factor > 1:
+            new_kwargs['repeats'] = cls.round_repeats(repeats, d_factor)
+
+        return new_kwargs
+
+
+class EfficientNetB0(ScalableModel):
+    """
+    EfficientNetB0
+
+    **Configuration**
+        see :class:`ScalableModel`
+        Base block is :meth:`.MobileNet_v2.block`
+
+    """
+
+    resolution = 224  # image resolution used in original paper
+
+    @classmethod
+    def default_config(cls):
+        config = super().default_config()
+
+        config['initial_block'].update(dict(scalable=True, layout='cna', kernel_size=3, strides=2, filters=32))
+
+        config['body'].update(dict(base=MobileNet_v2.block, blocks=[
```

Comment:
Don't you think that dictionary with parameters, each of which is `list`, would look nicer? You would be able to slice them just like `EncoderDecoder` does, and eliminate the need to use list of dictionaries with huge redundancy in name-typing

Reply:
This won't work for blocks that have different bases